### DISCUSSION TITLE
Utoipa with complex fields

### DISCUSSION THREAD

**Discussion by squadgazzz** [juhaku/utoipa#887] [Q&A]
Hello, from the provided examples, it is not clear how to use `utoipa` in the following scenario:
```
#[serde_as]
#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Solution {
    pub id: u64,
    #[serde_as(as = "HashMap<_, HexOrDecimalU256>")]
    pub prices: HashMap<H160, U256>,
}
```

Current API doc, which I am trying to regenerate with `utoipa`:
```
    Solution:
      description: |
        A computed solution for a given auction.
      type: object
      required:
        - id
        - prices
      properties:
        id:
          type: number
        prices:
          description: |
            A clearing price map of token address to price. The price can have
            arbitrary denomination.
          type: object
          additionalProperties:
            $ref: "#/components/schemas/U256"
            
    U256:
      description: |
        256 bit unsigned integer in decimal notation.
      type: string
      example: "1234567890"
```

Both `H160` and `U256` are external types. What would be the best way to create a schema for the `HashMap<H160, U256>` type without changing the prices field type(if possible)? I saw that wrappers are required, which would change the field type. Does any better workaround exist?

**Comment by juhaku** [comment]
> Both H160 and U256 are external types. What would be the best way to create a schema for the HashMap<H160, U256> type without changing the prices field type(if possible)? I saw that wrappers are required, which would change the field type. Does any better workaround exist?

At the moment the there are few ways to handle the external types. You can take a look at this comment and follow the links further for examples. https://github.com/juhaku/utoipa/issues/790#issuecomment-1787754185

Hope this helps. And sorry for late reply, At the moment I am overwhelmed by life and I have literally no time nor working dev setup to maintain the project. :disappointed:
